src/main.py

```diff
@@ -29,9 +30,27 @@
 import config
 import pubsub
 
+logging.config.dictConfig({
+    'version': 1,
+    'formatters': {'default': {
+        'format': '%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+    }},
+    'handlers': {'console': {
+        'class': 'logging.StreamHandler',
+        'stream': 'ext://sys.stderr',
+        'formatter': 'default'
+    }},
+    'root': {
+        'handlers': ['console']
+    }
+})
 
 app = Flask(__name__)
 app.config.from_object(config.load())
+
+root_logger = logging.getLogger()
+root_logger.setLevel(app.config["LOGGING_LEVEL"])
+app.logger.setLevel(app.config["LOGGING_LEVEL"])
```

Comment:
Sorry, I've been doing some more reading:

https://www.machinelearningplus.com/python/python-logging-guide/
https://flask.palletsprojects.com/en/1.1.x/logging/
https://stackoverflow.com/questions/27775026/provide-extra-information-to-flasks-app-logger

Would it be better here to do something like this:
```
# main.py

app = Flask(__name__)
app.config.from_object(config.load())
logging.basicConfig(level=app.config.LOGGING_LEVEL)

def some_function():
     app.logger.info('my message')
```
```
# some_other_module.py

def foo():
   logging.getLogger(__name__).info('my message')
```

Which I think should work? It sounds like basicConfig creates the root logger singleton, and sets the level. Then, the first time that someone calls `app.logger` Flask creates a logger that inherits from the root (so the level is inherited), and also inherits the flask app name (in this case, `__name__`).

Then, in other modules, when you call `logging.getLogger(__name__)`, this does a few things:
1. Inherits from the root logger. Because Flask also inherited from the root logger, it ensures that they don't diverge
2. Uses the default logging library, as opposed to custom Flask machinery. So then modules like pubsub.py aren't dependent on Flask (they can be called from non-flask applications).
3. Using `__name__` ensures that messages are logged to the correct namespace. Since the Flask app is also called `__name__`, it ensures that the two are consistent. 

I tested this in the REPL, but you might want to double check my work in the actual application before committing code. What do you think of this proposed logging system?

Reply:
I tested what you said in REPL and it worked for me, but when testing on the actual application (running on my local machine) it did not. It seemed as though the flask logger inherited the logging handler from the root logger, but did not inherit / share the level with the root logger. More specifically, I found that when I set the root logger level to WARNING, the flask logger still continues to log INFO messages. My speculation is that this is because the Flask app uses the logger while it is being initialized prior to the level of the root logger being set resulting in it having its own level.

I had a similar problem with the flask logger not directly inheriting the root's handler and logging format which is why I chose  to setup the logging configuration with dictConfig prior to initializing the flask application.

On the other hand, I just tested locally and found that child loggers created with ` logging.getLogger(__name__) ` did inherit the level from root, so I think this would definitely be a useful practice to use in the future.

Reply:
Thanks for the suggestions, I just made the modifications. While I was making the changes, I found that I continued to encounter the same problem of the flask logger not inheriting the logging level of the root logger. Although, after debugging some more, I found that the true reason was that the DevConfig had DEBUG set to True, and this resulted in the logger level switching when the config was loaded. I just tested it on CloudRun where DEBUG is false and everything is working as expected now.

Reply:
Thanks for figuring this out! To be clear, I think it's still okay to mediate access to the logger via `app.logger` in modules that make Flask imports. My main concern was keeping logging consistent between modules that import Flask, and modules that don't.  But, I also don't think that there's anything wrong with the code as is.